File: csharp/Source/Vetores/MaisVelho.cs
namespace Source.Vetores;

/**
 * Problema "mais_velho"
 * Fazer um programa para ler um conjunto de nomes de pessoas e suas respectivas idades.
 * Os nomes devem ser armazenados em um vetor, e as idades em um outro vetor.
 * Depois, mostrar na tela o nome da pessoa mais velha.
 */
internal static class MaisVelho
{
    private static readonly short MIN_SIZE = 1;
    private static readonly short MAX_SIZE = 10;

    public static void Execute()
    {
        Console.Write("Quantos pessoas você vai digitar (máximo de 10)? ");
        int n = int.Parse(Console.ReadLine()!);

        n = n < MIN_SIZE || n > MAX_SIZE ? MIN_SIZE : n;

        int[] ages = new int[n];
        string[] names = new string[n];

        for (int i = 0; i < n; i++)
        {
            Console.WriteLine($"Dados da {i + 1}ª pessoa de {n}:");

            Console.Write("Nome: ");
            names[i] = Console.ReadLine()!;

            Console.Write("Idade: ");
            ages[i] = int.Parse(Console.ReadLine()!);
        }

        int olderAgeIndex = 0;
        int olderAge = ages[olderAgeIndex];

        for (int i = 1; i < n; i++)
        {
            if (ages[i] > olderAge)
            {
                olderAge = ages[i];
                olderAgeIndex = i;
            }
        }

        Console.WriteLine($"Pessoa mais velha: {names[olderAgeIndex]}");
    }
}

File: python/src/vetores/mais_velho.py
""" Problema "mais_velho"
Fazer um programa para ler um conjunto de nomes de pessoas e suas respectivas idades.
Os nomes devem ser armazenados em um vetor, e as idades em outro vetor.
Depois, mostrar na tela o nome da pessoa mais velha.
"""

MIN_SIZE = 1
MAX_SIZE = 10


def mais_velho() -> None:
    n = int(input(f"Quantas pessoas você vai digitar (mínimo de {MIN_SIZE} e máximo de {MAX_SIZE})? "))

    if not MIN_SIZE < n <= MAX_SIZE:
        n = MIN_SIZE
    
    ages = n * [0]
    names = n * [""]

    for i in range(n):
        print(f"Dados da {i + 1}ª pessoa de {n}:")
        names[i] = input("Nome: ")
        ages[i] = int(input("Idade: "))

    older_age_index = 0
    older_age = ages[older_age_index]

    for i in range(1, n):
        if ages[i] > older_age:
            older_age = ages[i]
            older_age_index = i

    print(f"Pessoa mais velha: {names[older_age_index]}")


if __name__ == "__main__":
    mais_velho()
